test/core/Safe.Signatures.spec.ts

```diff
@@ -456,7 +449,176 @@ describe("Safe", () => {
+                signers: [user1, user2, user3],
+            } = await setupTests();
+            const safe = await getSafeWithOwners(
+                [user1.address, user2.address, user3.address],
+                3,
+                await compatFallbackHandler.getAddress(),
+            );
+            const safeAddress = await safe.getAddress();
+            const safeWithCompatFbHandlerIface = await getCompatFallbackHandler(safeAddress);
+            const tx = buildSafeTransaction({ to: safeAddress, nonce: await safe.nonce() });
+            const txHashData = preimageSafeTransactionHash(safeAddress, tx, await chainId());
+            const txHash = calculateSafeTransactionHash(safeAddress, tx, await chainId());
+            const signatures = buildSignatureBytes([
+                await safeApproveHash(user1, safe, tx),
+                await safeSignTypedData(user1, safeAddress, tx),
+                await safeSignTypedData(user3, safeAddress, tx),
+            ]);
+            await expect(safeWithCompatFbHandlerIface.checkSignatures(txHash, txHashData, signatures)).to.be.revertedWith("GS026");
+        });
+
+        it("should be able to mix all signature types", async () => {
+            const {
+                compatFallbackHandler,
+                signers: [user1, user2, user3, user4, user5],
+            } = await setupTests();
+            const compatFallbackHandlerAddress = await compatFallbackHandler.getAddress();
+            const signerSafe = await getSafeWithOwners([user5.address], 1, compatFallbackHandlerAddress);
+            const signerSafeAddress = await signerSafe.getAddress();
+            const safe = await getSafeWithOwners(
+                [user1.address, user2.address, user3.address, user4.address, signerSafeAddress],
+                5,
+                compatFallbackHandlerAddress,
+            );
+            const safeAddress = await safe.getAddress();
+            const tx = buildSafeTransaction({ to: safeAddress, nonce: await safe.nonce() });
+            const txHash = calculateSafeTransactionHash(safeAddress, tx, await chainId());
+            const safeWithCompatFbHandlerIface = await getCompatFallbackHandler(safeAddress);
+
+            const safeMessageHash = calculateSafeMessageHash(signerSafeAddress, txHash, await chainId());
+            const signerSafeOwnerSignature = await signHash(user5, safeMessageHash);
+            const signerSafeSig = buildContractSignature(signerSafeAddress, signerSafeOwnerSignature.data);
+
+            const signatures = buildSignatureBytes([
+                await safeApproveHash(user1, safe, tx, true),
+                await safeApproveHash(user4, safe, tx),
+                await safeSignTypedData(user2, safeAddress, tx),
+                await safeSignTypedData(user3, safeAddress, tx),
+                signerSafeSig,
+            ]);
+
+            await safeWithCompatFbHandlerIface.checkSignatures(txHash, "0x", signatures);
         });
```

Comment:
Might be worth adding a test for documentation purposes for "what happens when `keccak(data) != dataHash`?". AFAIU, it ignores it ATM, but I would still add a test to document that the behaviour is intended.

Reply:
I think the existing tests show that already don't they? `keccak("")` would always equal to `0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470` and never equal the txHash

Reply:
> My rationale is that if this is implied from the other test, you can change the behaviour at a later point in time without noticing it was intentional.

That would make all the current tests fail, wouldn't it? I'm fine with adding an explicit test, but to me, it feels more like a tautology (if the parameter is not used, it's implied to me that the hash of that parameter doesn't matter)

Reply:
That's fair. Just wanted to ask the question.